tools/shared.py

```diff
@@ -1167,6 +1165,9 @@ def load(self, args=[]):
           declare = re.sub(r'([\w\d]+)\s*=\s*(.+)', r'self.attrs["\1"]=\2;', args[i+1])
           exec declare
 
+      if get_llvm_target() == WASM_TARGET:
```

Comment:
i like this idea, but does this line occur early enough, i.e. before all the possible uses?

for maximal confidence it would be nice to add an assert on reading `Settings.WASM_BACKEND` that it has been initialized before. (we do something conceptually similar for `LLVM_TARGET`, set it to none, then in the getter, check it was set properly, but we'd need to use some python magic in this case I think)